invenio_records_resources/services/records/queryparser/query.py

```diff
@@ -82,13 +118,18 @@ def parse(self, query_str):
             # We parse the Lucene query syntax in Python, so we know upfront
             # if the syntax is correct before executing it in the search engine
             tree = luqum_parser.parse(query_str)
-
             # Perform transformation on the abstract syntax tree (AST)
-            if self.tree_transformer_factory is not None:
-                transformer = self.tree_transformer_factory()
+            if self.tree_transformer_cls is not None:
+                transformer = self.tree_transformer_cls(
+                    mapping=self.mapping, allow_list=self.allow_list
+                )
                 new_tree = transformer.visit(tree, context={"identity": self.identity})
                 query_str = str(new_tree)
             return dsl.Q("query_string", query=query_str, **self.extra_params)
         except (ParseError, QuerystringValidationError):
             # Fallback to a multi-match query.
+            if self.fields:
```

Comment:
Why this `if self.fields`?

Reply:
I've refactored a bit the code to try to make it more understandable and included a comment in the code:

>  in the case that there is no allow list nor fields in the extra_params we do not enter this if to avoid passing `fields=None`, however, if it exists it must overwrite a potential value given by the query to include the allow_list

It is related to this comment/answer too, https://github.com/inveniosoftware/invenio-records-resources/pull/465#discussion_r1162537906